### PR TITLE
[ja] Deep Learning

### PR DESCRIPTION
**Pull request by taixhi**
work in progress.

### CODE REVIEW

**Comment by taixhi** [comment]
The first draft is done. I would love for someone to have a look at it.

**Comment by shervinea** [comment]
Thank you for your translation @taixhi! If you know a native speaker to review your work, please feel free to spread the word about this PR.

**Comment by taniokah** [comment]
I would like to check this ja version today, and will comment here.

**Comment by taixhi** [comment]
Hi @taniokah, let me know if you have had a look at it :)

File: ja/cheatsheet-deep-learning.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+**53. [Recurrent Neural Networks, Gates, LSTM]**
+
+&#10230; [再帰型ニューラルネットワーク, ゲート, LSTM]
+
+<br>
+
+**54. [Reinforcement learning, Markov decision processes, Value/policy iteration, Approximate dynamic programming, Policy search]**
+
+&#10230; [強化学習, マルコフ決定過程, バリュー/ポリシー反復, 近似動的計画法, ポリシーサーチ]
```
Comment (Harimus) [inline ja/cheatsheet-deep-learning.md]: バリュー/ポリシー反復, -> 価値/方策反復

File: ja/cheatsheet-deep-learning.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+**52. [Convolutional Neural Networks, Convolutional layer, Batch normalization]**
+
+&#10230; [畳み込みニューラルネットワーク, 畳み込み層, バッチノーマライゼーション]
+
+<br>
+
+**53. [Recurrent Neural Networks, Gates, LSTM]**
+
+&#10230; [再帰型ニューラルネットワーク, ゲート, LSTM]
```
Comment (Harimus) [inline ja/cheatsheet-deep-learning.md]: See first comment about: 回帰型ニューラルネットワーク

File: ja/cheatsheet-deep-learning.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+**41. Bellman equation ― The optimal Bellman equations characterizes the value function Vπ∗ of the optimal policy π∗:**
+
+&#10230; ベルマン方程式 - 政策πをとった価値関数Vπ∗に対する最適なベルマン方程式：
+
+<br>
+
+**42. Remark: we note that the optimal policy π∗ for a given state s is such that:**
+
+&#10230; 備考: 与えられた状態sに対する最適方針π*はこのようになります：
```
Comment (Harimus) [inline ja/cheatsheet-deep-learning.md]: 方針 and 政策 is mixed in the translations. I'd suggest to use either 方策　or 方針 and stick with it.

**Comment by taniokah** [comment]
> Hi @taniokah, let me know if you have had a look at it :)

Sorry, my hand was stopped...
I will check the whole content in today 29/10/2019 JST.

**Comment by shervinea** [comment]
Hi @taixhi, please feel free to incorporate current suggestions to your work if you feel it would be appropriate. Looking forward to seeing the great work of this PR merged!

**Comment by taixhi** [comment]
oh wow, completely forgot about this, will take a look at in the next couple of days!

File: ja/cheatsheet-deep-learning.md [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+
+**8. Activation function ― Activation functions are used at the end of a hidden unit to introduce non-linear complexities to the model. Here are the most common ones:**
+
+&#10230; 活性化関数　ー ユニットの出力に非線形性を与える関数を活性化関数といいます。一般的には以下の関数がよく使われます：
+
+<br>
+
+**9. [Sigmoid, Tanh, ReLU, Leaky ReLU]**
+
+&#10230; [Sigmoid(シグモイド関数), Tanh(双曲線関数), ReLU(ランプ関数), Leaky ReLU]
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; [Sigmoid(シグモイド関数), Tanh(双曲線関数), ReLU(正規化線形ユニット), Leaky ReLU(漏洩ReLU)]
```

File: ja/cheatsheet-deep-learning.md [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**10. Cross-entropy loss ― In the context of neural networks, the cross-entropy loss L(z,y) is commonly used and is defined as follows:**
+
+&#10230; 交差エントロピーロス　ー ニューラルネットにおいて交差エントロピーロスL(z,y)は頻繁に使われ、以下のように定義されています：
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 交差エントロピーロス　ー ニューラルネットにおいて交差エントロピーロスL(z,y)は一般的に使われ、以下のように定義されています：
```

File: ja/cheatsheet-deep-learning.md [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+
+**43. Value iteration algorithm ― The value iteration algorithm is in two steps:**
+
+&#10230; 価値反復法アルゴリズム - 価値反復法アルゴリズムは２段階で行われます：
+
+<br>
+
+**44. 1) We initialize the value:**
+
+&#10230; 1) 値を初期化する。
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 1) 値を初期化する：
```

File: ja/cheatsheet-deep-learning.md [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+
+**38. Policy ― A policy π is a function π:S⟶A that maps states to actions.**
+
+&#10230; 政策 - 政策πは状態と行動を写像する関数π:S⟶A
+
+<br>
+
+**39. Remark: we say that we execute a given policy π if given a state s we take the action a=π(s).**
+
+&#10230; 備考: 状態sを与えられた際に行動a=π(s)を行うことを政策πを実行すると言う。
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 備考: 状態sを与えられた際に行動a=π(s)を行うことを方策πを実行すると言います。
```

File: ja/cheatsheet-deep-learning.md [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**40. Value function ― For a given policy π and a given state s, we define the value function Vπ as follows:**
+
+&#10230; 価値関数 - ある政策πとある状態sにおいて価値関数Vπを以下のように定義する：
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 価値関数 - ある方策πとある状態sにおける価値関数Vπを以下のように定義します：
```

File: ja/cheatsheet-deep-learning.md [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**41. Bellman equation ― The optimal Bellman equations characterizes the value function Vπ∗ of the optimal policy π∗:**
+
+&#10230; ベルマン方程式 - 政策πをとった価値関数Vπ∗に対する最適なベルマン方程式：
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; ベルマン方程式 - 最適なベルマン方程式は最適な方策π∗の価値関数Vπ∗とみなすことができます：
```

File: ja/cheatsheet-deep-learning.md [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+
+**10. Cross-entropy loss ― In the context of neural networks, the cross-entropy loss L(z,y) is commonly used and is defined as follows:**
+
+&#10230; 交差エントロピーロス　ー ニューラルネットにおいて交差エントロピーロスL(z,y)は頻繁に使われ、以下のように定義されています：
+
+<br>
+
+**11. Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. This can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.**
+
+&#10230; 学習率　ー αやηで表される学習率は勾配法による重み付けのアップデートをする速度を表します。学習率は固定または適応的に変更することができます。現在一般的に使われている学習法はAdam（アダム）であり、学習率を適用させる方法です。
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 学習率　ー 学習率は多くの場合α、しばしばηで表記され、勾配法による重み付けのアップデートをする速度を表します。学習率は固定または適応的に変更することができます。現在一般的に使われている学習法はAdam（アダム）であり、学習率を適用的に変更する方法です。
```

File: ja/cheatsheet-deep-learning.md (hunk 3 above)
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 備考: 与えられた状態sに対する最適方策π*はこのようになります：
```

File: ja/cheatsheet-deep-learning.md [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+
+**37. R:S×A⟶R or R:S⟶R is the reward function that the algorithm wants to maximize**
+
+&#10230; R:S×A⟶R or R:S⟶Rはアルゴリズムが最大化したい報酬関数
+
+<br>
+
+**38. Policy ― A policy π is a function π:S⟶A that maps states to actions.**
+
+&#10230; 政策 - 政策πは状態と行動を写像する関数π:S⟶A
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 方策 - 方策πは状態を行動に写像する関数π:S⟶A
```

File: ja/cheatsheet-deep-learning.md [hunk 12]
```diff
@@ -0,0 +1,321 @@
...
+
+**11. Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. This can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.**
+
+&#10230; 学習率　ー αやηで表される学習率は勾配法による重み付けのアップデートをする速度を表します。学習率は固定または適応的に変更することができます。現在一般的に使われている学習法はAdam（アダム）であり、学習率を適用させる方法です。
+
+<br>
+
+**12. Backpropagation ― Backpropagation is a method to update the weights in the neural network by taking into account the actual output and the desired output. The derivative with respect to weight w is computed using chain rule and is of the following form:**
+
+&#10230; 誤差逆伝播法（backpropagation）ー 誤差逆伝播法はニューラルネットの期待される出力値と実際の出力の差異を考慮し重み付けのアップデートをする方法の一つです。重みwに関する導関数は連鎖規則を使用して計算され、次の形式で表される：
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 誤差逆伝播法（backpropagation）ー 誤差逆伝播法はニューラルネットにおいて実際の出力と期待される出力との差異を考慮して重みを更新する方法の一つです。重みwに関する導関数は連鎖律を使用して計算され、次の形式で表されます：
```

File: ja/cheatsheet-deep-learning.md [hunk 13]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**13. As a result, the weight is updated as follows:**
+
+&#10230; 結果、重みは以下のようにアップデートされます：
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 結果として、重みは以下のように更新されます：
```

File: ja/cheatsheet-deep-learning.md [hunk 14]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**14. Updating weights ― In a neural network, weights are updated as follows:**
+
+&#10230; 重みアップデート　ー ニューラルネットでは以下のように重みがアップデートされます：
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 重みの更新 ー ニューラルネットでは以下のように重みが更新されます：
```

File: ja/cheatsheet-deep-learning.md [hunk 15]
```diff
@@ -0,0 +1,321 @@
...
+
+**35. {Psa} are the state transition probabilities for s∈S and a∈A**
+
+&#10230; {Psa}は状態s∈Sと行動a∈Aの状態遷移確率
+
+<br>
+
+**36. γ∈[0,1[ is the discount factor**
+
+&#10230; γ∈[0,1[は割引因子と呼ばれる値
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; γ∈[0,1[は割引因子
```

File: ja/cheatsheet-deep-learning.md [hunk 16]
```diff
@@ -0,0 +1,321 @@
...
+
+**15. Step 1: Take a batch of training data.**
+
+&#10230; ステップ１： 訓練データを１バッチ用意する。
+
+<br>
+
+**16. Step 2: Perform forward propagation to obtain the corresponding loss.**
+
+&#10230; ステップ２：　フォワードプロパゲーションを行い誤差を求める。
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; ステップ２：　順伝播を行いそれに対する誤差を求める。
```

File: ja/cheatsheet-deep-learning.md [hunk 17]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**17. Step 3: Backpropagate the loss to get the gradients.**
+
+&#10230; 求められた誤差を用い、傾斜を計算する。
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 誤差を逆伝播し、勾配を計算する。
```

File: ja/cheatsheet-deep-learning.md [hunk 18]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**18. Step 4: Use the gradients to update the weights of the network.**
+
+&#10230; 傾斜を使い誤差が小さくなるように重みを調整する。
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 勾配を使いネットワークの重みを更新する。
```

File: ja/cheatsheet-deep-learning.md [hunk 19]
```diff
@@ -0,0 +1,321 @@
...
+
+<br>
+
+**19. Dropout ― Dropout is a technique meant at preventing overfitting the training data by dropping out units in a neural network. In practice, neurons are either dropped with probability p or kept with probability 1−p**
+
+&#10230;ドロップアウト　ー ドロップアウトはニューラルネット内の一部のユニットを非活性化させることにより過学習を防ぐテクニックである。実際には、ニューロンはある確率pで非活性、1-pの確率で活性化されるようになってる。
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230;ドロップアウト　ー ドロップアウトはニューラルネット内の一部のユニットを無効にすることで学習データへの過学習を防ぐテクニックです。実際には、ニューロンは確率pで無効、確率1-pで有効のどちらかになります。
```

File: ja/cheatsheet-deep-learning.md [hunk 20]
```diff
@@ -0,0 +1,321 @@
...
+
+**20. Convolutional Neural Networks**
+
+&#10230; 畳み込みニューラルネットワーク
+
+<br>
+
+**21. Convolutional layer requirement ― By noting W the input volume size, F the size of the convolutional layer neurons, P the amount of zero padding, then the number of neurons N that fit in a given volume is such that:**
+
+&#10230; 畳み込みレイヤーの条件　ー Wを入力サイズ、Fを畳み込みレイヤーニューロンのサイズ、Pをゼロパディングの量とすると、与えられた体積に収まるニューロン数Nは次のようになります。
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 畳み込みレイヤーの条件　ー Wを入力サイズ、Fを畳み込み層のニューロンのサイズ、Pをゼロ埋めの量とすると、これらに対応するニューロンの数Nは次のようになります。
```

File: ja/cheatsheet-deep-learning.md [hunk 21]
```diff
@@ -0,0 +1,321 @@
...
+
+**48. times took action a in state s**
+
+&#10230; 状態sで行動aを行った回数
+
+<br>
+
+**49. Q-learning ― Q-learning is a model-free estimation of Q, which is done as follows:**
+
+&#10230;  Q学習　ー Q学習は数学モデルを使わないQ値の評価手法であり、以下のように行われる：
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230;  Q学習　ー Q学習はモデルフリーのQ値の推定であり、以下のように行われます：
```

File: ja/cheatsheet-deep-learning.md [hunk 22]
```diff
@@ -0,0 +1,321 @@
...
+
+**21. Convolutional layer requirement ― By noting W the input volume size, F the size of the convolutional layer neurons, P the amount of zero padding, then the number of neurons N that fit in a given volume is such that:**
+
+&#10230; 畳み込みレイヤーの条件　ー Wを入力サイズ、Fを畳み込みレイヤーニューロンのサイズ、Pをゼロパディングの量とすると、与えられた体積に収まるニューロン数Nは次のようになります。
+
+<br>
+
+**22. Batch normalization ― It is a step of hyperparameter γ,β that normalizes the batch {xi}. By noting μB,σ2B the mean and variance of that we want to correct to the batch, it is done as follows:**
+
+&#10230; バッチ正規化　ー バッチ{xi}を正規化するハイパーパラメータγ、βのステップです。バッチに修正したい平均値と分散値をμB,σ2Bとすると、正規化は以下のように行われます:
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; バッチ正規化　ー バッチ{xi}を正規化するハイパーパラメータγ、βのステップです。補正を行うバッチの平均と分散をμB,σ2Bとすると、正規化は以下のように行われます:
```

File: ja/cheatsheet-deep-learning.md [hunk 23]
```diff
@@ -0,0 +1,321 @@
...
+
+**23. It is usually done after a fully connected/convolutional layer and before a non-linearity layer and aims at allowing higher learning rates and reducing the strong dependence on initialization.**
+
+&#10230; これは通常、学習率を高め、初期値への依存性を減らすことを目的でFully Connected層と畳み込み層の後、非線形化を行う前に行われます。
+
+<br>
+
+**24. Recurrent Neural Networks**
+
+&#10230; 再帰型ニューラルネットワーク (RNN)
```
Comment (yoshiyukinakai) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; リカレントニューラルネットワーク (RNN)
```

File: ja/cheatsheet-deep-learning.md [hunk 24]
```diff
@@ -0,0 +1,321 @@
...
+
+**22. Batch normalization ― It is a step of hyperparameter γ,β that normalizes the batch {xi}. By noting μB,σ2B the mean and variance of that we want to correct to the batch, it is done as follows:**
+
+&#10230; バッチ正規化　ー バッチ{xi}を正規化するハイパーパラメータγ、βのステップです。バッチに修正したい平均値と分散値をμB,σ2Bとすると、正規化は以下のように行われます:
+
+<br>
+
+**23. It is usually done after a fully connected/convolutional layer and before a non-linearity layer and aims at allowing higher learning rates and reducing the strong dependence on initialization.**
+
+&#10230; これは通常、学習率を高め、初期値への依存性を減らすことを目的でFully Connected層と畳み込み層の後、非線形化を行う前に行われます。
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; これは通常、学習率を高め、初期値への強い依存性を減らすことを目的として、全結合層もしくは畳み込み層の後、非線形化層の前で行われます。
```

File: ja/cheatsheet-deep-learning.md [hunk 25]
```diff
@@ -0,0 +1,321 @@
...
+
+**28. LSTM ― A long short-term memory (LSTM) network is a type of RNN model that avoids the vanishing gradient problem by adding 'forget' gates.**
+
+&#10230; LSTM - A long short-term memory (LSTM) ネットワークは勾配消失問題を解決するために忘却ゲートが追加されているRNNの一種です。
+
+<br>
+
+**29. Reinforcement Learning and Control**
+
+&#10230; 強化学習とコントロール
```
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; 強化学習と制御
```

File: ja/cheatsheet-deep-learning.md (hunk 9 above)
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; ベルマン方程式 - 最適ベルマン方程式は最適方策π∗の価値関数Vπ∗で記述されます：
```

File: ja/cheatsheet-deep-learning.md (hunk 1 above)
Comment (for-tokyo) [inline ja/cheatsheet-deep-learning.md]:
```suggestion
&#10230; [強化学習, マルコフ決定過程, 価値/方策反復, 近似動的計画法, 方策探索]
```

**Comment by yoshiyukinakai** [comment]
Hello @taixhi, a team from Machine Learning Tokyo completed reviewing your translation and added some suggestions. Could you check and incorporate our suggestions?

Here is how to incorporate suggestions:
https://help.github.com/ja/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

**Comment by yoshiyukinakai** [comment]
Hello @taixhi, this is a friendly reminder that we completed reviewing your pull request. Could you take a look at our suggestions?

**Comment by taixhi** [comment]
Sorry about the delay guys, just reviewed... Looks good to me! It was a good learning opportunity for me as well :)

**Comment by yoshiyukinakai** [comment]
Hello @shervinea, we completed translation and review. Could you check if you can merge this pull request?

**Comment by shervinea** [comment]
Thank you @yoshiyukinakai and everyone else who contributed to this translation. Your work is really appreciated! I'll go ahead and proceed to the merge as well as fill out the `CONTRIBUTORS` file accordingly.

**Pull request merged**